static/js/main.js

```diff
@@ -534,6 +536,46 @@ function getUrlVars() {
    return vars;
 }
 
+function autoCompleteLookup(query, done)  {
+   // Only trigger lookup for certain prefixes
+   var none = { 'suggestions': [] };
+   var whois = 'Hver er ';
+   var whatis = 'Hvað er ';
+   var minqlen = whois.length + 1;
+   var valid = (query.startsWith(whois) || query.startsWith(whatis)) &&
+               query.length >= minqlen && !query.endsWith('?');
+   if (!valid) {
+      done(none);
+      return;
+   }
+   // Local caching
+   if (autoCompleteLookup.cache === undefined) {
+      autoCompleteLookup.cache = { };
+   }
+   cache = autoCompleteLookup.cache;
+   if (cache[query] !== undefined) {
+      done(cache[query]);
+      return;
+   }
+   // Cancel any active request
+   if (autoCompleteLookup.req) {
+      autoCompleteLookup.req.abort();
+   }
+   // Ajax request to server
+   autoCompleteLookup.req = $.ajax({
+      type: 'GET',
+      url: "/suggest?q="+ query,
+      dataType: "json",
+      success: function(json) {
+         autoCompleteLookup.cache[query] = json;
+         done(json);
+      },
+      error: function(ajaxContext) {
+         console.log(ajaxContext.responseText)

```

Comment:
What does the UI show in the case of an error here?

Reply:
Nothing. Autocompletion of queries is a nice-to-have, non-essential feature. If the request fails, the suggestions simply don't appear. To my mind, it makes no sense to notify the user client-side.